### ISSUE TITLE
Games always run on Nvidia GPU Offload

### ISSUE THREAD

**Issue by UnicodeMan** [Heroic-Games-Launcher/HeroicGamesLauncher#655]
I have a laptop with internal GPU Intel UHD 630 + external Nvidia GTX1050. 

Games always run on Nvidia GPU, doesn't matter if I turn the knob for prime render or don't.
I think this is because of dxvk/vulkan, and should be fixed if one could select VULKAN icd loaders.
By selecting Intel icd loader explicitly to run the game on Intel GPU, issue should be resolved.

Same as in Lutris, because selecting icd loader always resolves that issue for DXVK games.
Not sure if the issue persists for OpenGL/other games on Heroic or on Lutris.

**Comment by flavioislima** [comment]
What happens if you pass the environment variables to load the ICD loaders?
Because I'm not sure how this works since I don't have an Optimus laptop for testing.

So do you want some games to run on the Intel GPU instead of the Nvidia, right?

**Comment by UnicodeMan** [comment]
In reverse order: 

1) Correct, I want to use Intel GPU instead of Nvidia (for some lighter games).

2) running "VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/intel_icd.x86_64.json heroic" in terminal allowed me to launch "Absolute Drift" in Intel GPU (mangohud showed 139mhz clock of Nvidia GPU and 0.0GB vram utilisation). 

3) some games crashed and did not launch at all with 64bit intel icd. This is similar to lutris - some games require 32bit Intel Vulkan icd to launch.

In conclusion - using Vulkan icd environment variable works.
Did not know that such variable existed, will use that in the meantime. :)

**Comment by Leopard1907** [comment]
https://github.com/doitsujin/dxvk/blob/master/src/dxvk/dxvk_instance.cpp#L183

DXVK does sort gpu devices and selects dgpu first if nothing is specified.

http://us.download.nvidia.com/XFree86/Linux-x86_64/470.74/README/primerenderoffload.html

You can either specify icd like you did above or pass the ```__NV_PRIME_RENDER_OFFLOAD=1 __VK_LAYER_NV_optimus=non_NVIDIA_only``` to make stuff work on non NV devices.

**Comment by UnicodeMan** [comment]
> You can either specify icd like you did above or pass the `__NV_PRIME_RENDER_OFFLOAD=1 __VK_LAYER_NV_optimus=non_NVIDIA_only` to make stuff work on non NV devices.

This workaround has no effect when specified in Heroic launcher's game-specific dedicated environment variable textbox. Only when you launch Heroic from terminal (when it is not running in the system) with mentioned env variables does it work.

A picture below to show what I meant by where exactly this has no effect.

![image](https://user-images.githubusercontent.com/25935211/135314062-336cf349-efab-48cd-a214-1cc616235dd8.png)

It is a shame because one may prefer to launch different games using different GPUs.

**Comment by UnicodeMan** [comment]
Oh wait... It actually does work there, however I have to pass all of the above:
`VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/intel_icd.x86_64.json __NV_PRIME_RENDER_OFFLOAD=1  __VK_LAYER_NV_optimus=non_NVIDIA_only`

**Comment by flavioislima** [comment]
Ok, I'll add this to our Wiki to help other people with the same problem, but I'll close the issue.
